### PR TITLE
Fix loader2 bug on 3dsmax 2024

### PR DESCRIPTION
**Pull request by carlos-villavicencio-adsk**
- Keep the reference of the dialog as an attribute of `self` to avoid deletion by the garbage collector
    - Reference: https://forum.qt.io/post/519764
- Update tests README
- Refactor run_tests

### CODE REVIEW

**Comment by carlos-villavicencio-adsk** [comment]
FYI all tests are broken, probably because we haven't maintained them for a while. We can work on this on a separate ticket.

image

File: tests/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 To run these tests, you need to do the following
 
-1. From a terminal, create a virtual environment at `tests/venv`.
+1. From a terminal, create a virtual environment at `tests/venv`. You need to use `virtualenv` package.
```
Comment (julien-lang) [inline tests/README.md:5]: virtualenv was for Py2. In Py3, it's venv

**Review by julien-lang — COMMENTED** [review]
@carlos-villavicencio-adsk yes, please create a ticket for the broken tests.

File: tests/README.md (hunk 1 above)
Comment (carlos-villavicencio-adsk) [reply tests/README.md:5]: You're right. However, the `run_tests.py` logic expects a virtual env created by `virtualenv` which still can be used on Python 3. The [script expects](https://github.com/shotgunsoftware/tk-3dsmax/blob/master/tests/run_tests.py#L20) the `activate_this.py` file to be part of the virtual environment created by `virtualenv`.

**Review by julien-lang — APPROVED** [review]

**Review by NorberMV — APPROVED** [review]
LGTM!

**Pull request merged**